<launch>
    <arg name="pcl_file" default="changeme.pcd"/>
    <arg name="z_threshold" default="0.0"/>
    <node pkg="rviz2" exec="rviz2" args="-d $(find-pkg-share orb_slam3_ros2)/config/point_cloud.rviz"/>
    <node pkg="orb_slam3_ros2" exec="visualize_point_cloud">
        <param name="map_file_name" value="$(var map_file_name)"/>
        <param name="pcl_file" value="$(var pcl_file)"/>
        <param name="z_threshold" value="$(var z_threshold)"/>
    </node>
    <node pkg="tf2_ros" exec="static_transform_publisher" args="--x 0 --y 0 --z 0 --yaw 0 --pitch 0 --roll 0 --frame-id world --child-frame-id map"/>

	<node pkg="octomap_server" exec="octomap_server_node" name="octomap_server">
		<param name="resolution" value="0.05" />

		<!-- fixed map frame (set to 'map' if SLAM or localization running!) -->
		<param name="frame_id" value="map" />

		<!-- maximum range to integrate (speedup!) -->
		<param name="sensor_model.max_range" value="5.0" />

        <param name="filter_ground_plane" value="true" />

		<!-- data source to integrate (PointCloud2) -->
		<remap from="cloud_in" to="orb_point_cloud2" />
	</node>
</launch>
